### PR TITLE
feat(network)!: distinguish Arcade from ARC; rewire GorillaPool provider (#775)

### LINKED ISSUE #776: Task: Refactor ARC broadcast escape hatches to read path from endpoint table
Parent HLR: #775
Analyst's refined breakdown: https://github.com/sgbett/bsv-ruby-sdk/issues/775#issuecomment-4575268489

## Dependencies

- **Depends on**: none (this is the first task in the chain)
- **Blocks**: #777, #779, #780, #781

## Scope

Decouple ARC's broadcast escape hatches from hardcoded paths so that sibling protocols (Arcade) can reuse the broadcast skeleton.

Lift `safe_parse_json` to a shared location callable from both `ARC` and the forthcoming `Arcade`. (`resolve_tx_hex` may also be lifted; rationale that it benefits `TAALBinary` was incorrect — `TAALBinary` does binary coercion, not hex. Lift is still warranted for Arcade reuse alone.)

## Files changed

- **New**: `gem/bsv-sdk/lib/bsv/network/util.rb` (or `BSV::Network::JSONHelpers` mixin module) — houses `safe_parse_json` (and `resolve_tx_hex` if lifted).
- **Modified**: `gem/bsv-sdk/lib/bsv/network/protocols.rb` — autoload entry for the new helper.
- **Modified**: `gem/bsv-sdk/lib/bsv/network/protocols/arc.rb`:
  - Replace hardcoded `'/v1/tx'` in `call_broadcast` with `self.class.endpoints[:broadcast][:path]` (currently at line 95).
  - Replace hardcoded `'/v1/txs'` in `call_broadcast_many` with `self.class.endpoints[:broadcast_many][:path]` (currently at line 127).
  - Delegate `safe_parse_json` (and `resolve_tx_hex` if lifted) to the shared module.

## Acceptance criteria

- [ ] `safe_parse_json` lives in a shared location callable by other protocols.
- [ ] No hardcoded broadcast paths in `lib/bsv/network/protocols/arc.rb`.
- [ ] Zero behaviour change; all existing `spec/bsv/network/protocols/arc_spec.rb` examples remain green without modification.
- [ ] `arc_integration_spec.rb` examples remain unchanged at this stage (rewired in Task #779).
- [ ] All existing specs pass (`bundle exec rake`).
- [ ] One additional unit test: subclass `ARC` with a custom `endpoint :broadcast, :post, '/custom/tx', response: :json` and verify the call routes to `/custom/tx`.

## Edge cases

- `safe_parse_json` must continue to return `{ 'detail' => nil }` (not `{ 'detail' => '' }`) on nil/empty input — covered by existing `arc_spec.rb` examples.
- Path lookup is safe with the current endpoint macro shape (`self.class.endpoints[:broadcast][:path]`).

## Out of scope

- Adding the new Arcade protocol (Task #777).
- Changing any ARC response semantics.

### PR DESCRIPTION
**Pull request by sgbett**
Closes HLR #775. Closes #776, #777, #779, #780, #781.
Leaves follow-up #778 open (wiring GorillaPool's chaintracks_server at its separate host/port).

## Summary

GorillaPool's `arcade.gorillapool.io` runs [`bsv-blockchain/arcade`](https://github.com/bsv-blockchain/arcade), not [`bitcoin-sv/arc`](https://github.com/bitcoin-sv/arc). Despite Arcade's marketing as "ARC-compatible", only some header names overlap — paths, response bodies, and status taxonomies all differ. As a result every `Providers::GorillaPool` broadcast, status, health, and chaintracks call was 404ing in production.

This PR models Arcade as a distinct first-class protocol (`Protocols::Arcade`), rewires `Providers::GorillaPool` to use it, removes the equally-broken `Protocols::Chaintracks` registration (follow-up at #778), and updates every downstream consumer.

## Why this slipped through

`spec/bsv/network/protocols/arc_integration_spec.rb` was hitting `arcade.gorillapool.io` and accepting 404 as a pass via `be_http_not_found.or(...)`. The defence has been removed and the spec re-pointed at `arc.taal.com`. A new env-gated live broadcast integration spec exercises both providers end-to-end so this category of regression cannot reach `master` again.

### New protocol
- `Protocols::Arcade` — `broadcast (POST /tx)`, `get_tx_status (GET /tx/{txid})`, `health (GET /health)`. Arcade-specific `REJECTED_STATUSES`, headers (`X-CallbackUrl`, `X-FullStatusUpdates`, etc.), and response parser handling `{status: "submitted"}` / `{status: "already submitted"}` / `400 {reason}` / `503 Retry-After` / 404-on-unknown-txid.

### Shared helpers lifted
- `safe_parse_json` and `resolve_tx_hex` moved to `BSV::Network::Util`. ARC and Arcade both delegate.
- ARC broadcast escape hatches now read their path from `self.class.endpoints[:broadcast][:path]` rather than hardcoding `/v1/tx` and `/v1/txs`.

### Provider rewire (breaking)
- `Providers::GorillaPool.mainnet` registers `Protocols::Arcade`, `Protocols::Ordinals`, `Protocols::JungleBus` (was `ARC`, `Chaintracks`, `Ordinals`, `JungleBus`).
- `Providers::GorillaPool.testnet` registers `Protocols::Arcade` only (was `ARC`, `Chaintracks`).
- Inline `# TODO: see #778` marks the chaintracks_server follow-up.

### Downstream consumer fixes
- `LivePolicy.default` switched to `https://arc.taal.com` (TAAL still runs ARC and serves `/v1/policy`); falls back on failure as before.
- `broadcast_p2pkh` MCP tool adapted to Arcade's `{status: "submitted"}` body shape; stays GorillaPool-pinned.
- `ChainTrackers::Chaintracks.default` holds the GorillaPool chaintracks URL constant locally; testnet raises `NotImplementedError` pointing at #778.

### Spec hardening
- `arc_integration_spec.rb` re-pointed to `arc.taal.com` with the `.or` defence removed.
- `auth_integration_spec.rb` GorillaPool scenarios reworked for Arcade.
- `defaults_spec.rb` updated for new protocol counts and ordering.
- `testnet_spec.rb` renamed `BSV_TESTNET_ARC_URL` → `BSV_TESTNET_ARCADE_URL`.
- New `broadcast_integration_spec.rb` — env-gated on `BSV_LIVE_TEST_WIF`, skips cleanly without funds, exercises both TAAL ARC and GorillaPool Arcade live.

### Docs & version
- `docs/network/overview.md` and `docs/network/examples.md` explain the ARC vs Arcade split.
- `docs/testing/testnet.md` corrects the env-var documentation.
- `CHANGELOG.md` entry under `0.21.0` flags Breaking Changes / Removed / Added / Changed.
- `lib/bsv/version.rb` bumped `0.20.0` → `0.21.0` (pre-1.0 minor bump per project convention; no shim).

## Verification

| Check | Result |
|---|---|
| `bundle exec rake` (full suite, both gems) | ✅ 5915 examples / 0 failures / 22 pre-existing pending |
| `bundle exec rubocop` (406 files) | ✅ no offenses |
| `Providers::GorillaPool.mainnet.protocols.length` | ✅ 3 (Arcade, Ordinals, JungleBus) |
| `Providers::GorillaPool.testnet.protocols.length` | ✅ 1 (Arcade) |
| `Protocols::Arcade.commands` | ✅ `Set[:broadcast, :get_tx_status, :health]` |
| No hardcoded `/v1/tx` or `/v1/txs` outside endpoint table | ✅ |
| `bsv-sdk` version | ✅ `0.21.0` |
| CHANGELOG entry under `0.21.0` | ✅ |
| Follow-up #778 referenced inline and in CHANGELOG | ✅ |

## Breaking changes

`Providers::GorillaPool.call(:broadcast, ...)` now returns Arcade response shape, not ARC. Status field is `data['status']` (e.g. `"submitted"`), not `data['txStatus']`. Pre-1.0 clean break — no compatibility shim. Consumers reading ARC-shape responses from GorillaPool must adapt.

`Providers::GorillaPool` no longer registers `Protocols::Chaintracks`. Callers relying on it through GorillaPool must construct their own Chaintracks protocol instance, or wait for #778. (Grep confirmed no internal callers.)

## Out of scope

- Wiring GorillaPool's `chaintracks_server` at its separate host/port — follow-up #778.
- Arcade `:broadcast_many`, `:events`, `:ready` (Arcade `:get_policy` does not exist upstream).
- Intent-based routing across providers.

## Test plan

- [x] `bundle exec rake` green
- [x] `bundle exec rubocop` green
- [x] All 8 acceptance criteria on HLR #775 verified against the code
- [ ] (Optional, reviewer) run live broadcast spec with `BSV_LIVE_TEST_WIF` set
- [ ] CI checks pass on this PR

## Release plan

After merge, run `/release sdk` to ship `bsv-sdk 0.21.0`. The CHANGELOG entry is already promoted to `## 0.21.0`.

### CODE REVIEW

File: gem/bsv-sdk/lib/bsv/network/util.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module BSV
```
[the remark this answers is not in the document]
Comment (sgbett) [reply gem/bsv-sdk/lib/bsv/network/util.rb:5]: Fixed in 4e475018 — added `require 'json'` to util.rb. Good catch; this file was implicitly relying on protocol.rb loading first.

File: gem/bsv-sdk/spec/bsv/network/protocols/broadcast_integration_spec.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    if prevout_env
+      parts = prevout_env.split(':')
+      raise 'BSV_LIVE_TEST_PREVOUT must be <dtxid_hex>:<vout>:<satoshis>' unless parts.length == 3
+
+      { 'tx_hash' => parts[0], 'tx_pos' => Integer(parts[1]), 'value' => Integer(parts[2]) }
+    else
+      woc = BSV::Network::Providers::WhatsOnChain.mainnet
+      result = woc.call(:get_utxos, address)
+      skip 'WoC rate-limited — set BSV_LIVE_TEST_PREVOUT to bypass' if result.http_response.code == '429'
+      raise "WoC UTXO fetch failed: #{result.error_message}" unless result.http_success?
```
[the remark this answers is not in the document]
Comment (sgbett) [reply gem/bsv-sdk/spec/bsv/network/protocols/broadcast_integration_spec.rb:38]: Fixed in 4e475018 — replaced with `result.code == '429'`. `ProtocolResponse` exposes `#code`, not `#http_response`.

File: gem/bsv-sdk/spec/bsv/network/protocols/broadcast_integration_spec.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    expect(result.data['txStatus']).to be_in(%w[SEEN_ON_NETWORK RECEIVED MINED])
+  end
+
+  it 'broadcasts the same tx via GorillaPool Arcade (idempotent re-broadcast)' do
+    taal.call(:broadcast, tx)
+
+    result = gorilla_pool.call(:broadcast, tx)
+    expect(result).to be_http_success
+    expect(result.data['status']).to be_in(['submitted', 'already submitted'])
+  end
```
[the remark this answers is not in the document]
Comment (sgbett) [reply gem/bsv-sdk/spec/bsv/network/protocols/broadcast_integration_spec.rb]: Fixed in 4e475018 — collapsed to a single end-to-end example that does both broadcasts and both status round-trips in sequence. Avoids the inconsistent-UTXO-state risk between examples.

File: gem/bsv-sdk/spec/bsv/network/protocols/broadcast_integration_spec.rb [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    expect(broadcast).to be_http_success
+
+    txid = tx.txid_hex
+    sleep 2
+
+    status = taal.call(:get_tx_status, txid)
+    expect(status).to be_http_success
+    expect(status.data['txid']).to eq(txid)
+    expect(status.data['txStatus']).to be_in(%w[RECEIVED SEEN_ON_NETWORK MINED])
+  end
```
[the remark this answers is not in the document]
Comment (sgbett) [reply gem/bsv-sdk/spec/bsv/network/protocols/broadcast_integration_spec.rb]: Fixed in 4e475018 — the collapsed example now exercises both providers end-to-end (TAAL ARC broadcast + status, GorillaPool Arcade broadcast + status).

File: gem/bsv-sdk/lib/bsv/transaction/fee_models/live_policy.rb [hunk 5]
```diff
@@ -34,15 +34,17 @@ class LivePolicy < FeeModel
...
+        # @param api_key [String, nil] optional TAAL API key for authenticated policy access
         # @return [LivePolicy]
         def self.default(api_key: nil)
-          provider = BSV::Network::Providers::GorillaPool.mainnet
-          arc_protocol = provider.protocol_for(:broadcast)
-          new(arc_url: arc_protocol.base_url, fallback_rate: DEFAULT_FALLBACK_RATE, api_key: api_key)
+          new(arc_url: TAAL_ARC_URL, fallback_rate: DEFAULT_FALLBACK_RATE, api_key: api_key)
         end
 
         # @param arc_url [String] ARC base URL (e.g. 'https://arcade.gorillapool.io')
```
[the remark this answers is not in the document]
Comment (sgbett) [reply gem/bsv-sdk/lib/bsv/transaction/fee_models/live_policy.rb]: Fixed in 4e475018 — updated the `@param arc_url` example to `https://arc.taal.com` to match the new `LivePolicy.default` URL.

File: docs/testing/testnet.md [hunk 6]
```diff
@@ -59,9 +59,9 @@ tx.add_output(payment)
 tx.add_output(change)
 tx.sign_all(private_key)
 
-response = arc.broadcast(tx)
-response.success?  # => true
-response.txid      # => "a1b2c3..."
+response = arcade.broadcast(tx)
+response.http_success?       # => true
+response.data['status']      # => "submitted"
```
[the remark this answers is not in the document]
Comment (sgbett) [reply docs/testing/testnet.md:64]: Fixed in 4e475018 — updated to `arcade.call(:broadcast, tx)` matching the actual protocol API.

File: gem/bsv-sdk/spec/bsv/network/protocols/all_protocols_spec.rb [hunk 7]
```diff
@@ -4,12 +4,16 @@
 require 'spec_helper'
 
 RSpec.describe 'BSV::Network::Protocols integration' do
-  # Verify all 5 protocol classes load correctly via autoload.
+  # Verify all 6 protocol classes load correctly via autoload.
```
[the remark this answers is not in the document]
Comment (sgbett) [reply gem/bsv-sdk/spec/bsv/network/protocols/all_protocols_spec.rb]: Fixed in 3ff01c31 — corrected the count to 7 (now matches the autoload list).

File: gem/bsv-sdk/spec/bsv/network/protocols/all_protocols_spec.rb [hunk 8]
```diff
@@ -4,12 +4,16 @@
...
       expect(BSV::Network::Protocols::ARC).to be < BSV::Network::Protocol
     end
 
+    it 'loads Arcade' do
+      expect(BSV::Network::Protocols::Arcade).to be < BSV::Network::Protocol
+    end
+
     it 'loads WoCREST' do
       expect(BSV::Network::Protocols::WoCREST).to be < BSV::Network::Protocol
     end
```
[the remark this answers is not in the document]
Comment (sgbett) [reply gem/bsv-sdk/spec/bsv/network/protocols/all_protocols_spec.rb]: Fixed in 3ff01c31 — added a JungleBus autoload smoke test alongside the others.
